packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx

```diff
@@ -232,6 +250,112 @@ export class FocusZone extends BaseComponent<IFocusZoneProps, {}> implements IFo
     }
   };
 
+  /**
+   * Callback hooked to the root element. When we receive a blur, we want
+   * to detect if the blur is due to an element being removed from the Zone.
+   * This is done by evaluating if the destination (in ev.relatedTarget) is
+   * null. If so, we need to move the focus to either the next element, or
+   * possibly, the previous element if there is no "next".
+   *
+   * In the case there are no more focusable elements in the zone, we "park"
+   * focus on the root temporarily. If the user tabs elsewhere, we remove
+   * focusablilty from the root. After a zone re-render, if focus is still
+   * "parked", we try to move focus back to the first focusable child.
+   *
+   * Note: Attempting to move this to a React event, rather than using a
+   * native event, broke the implementation.
+   */
+  private _onBlurCapture(ev: FocusEvent) {
+    const { current: root } = this._root;
+    const { target: from } = ev;
+    const isFocusDisappearing = ev.relatedTarget === null || ev.relatedTarget === document;
+
+    // If we are tabbing from parked state, reset parked.
+    if (from === root) {
+      this._setParkedFocus(false);
+    }
+
+    if (isFocusDisappearing) {
+      // The element may be removed; we need remember its index path and asynchronously
+      // re-evaluate if the item was removed. Blurs can also happen if the user just
+      // clicks the location bar, so we need to disambiguate.
+      const indexPath = this._getIndexPath(from as HTMLElement);
+
+      this._async.setTimeout(() => {
+        if (!elementContains(root, from as HTMLElement)) {
+          const nextFocusedElement = this._getElementByIndexPath(indexPath);
+
+          if (nextFocusedElement) {
+            // Try to move focus to the next one.
+            this._setActiveElement(nextFocusedElement, true);
+            nextFocusedElement.focus();
+          } else {
+            // There are no more elements. Park focus on the container.
+            this._setParkedFocus(true);
+          }
+        }
+      }, 0);
+    }
+  }
+
+  private _setParkedFocus(isParked: boolean): void {
+    const { current: root } = this._root;
+    if (root) {
+      if (isParked) {
+        if (!this.props.allowFocusRoot) {
+          this._parkedTabIndex = root.getAttribute('tabindex');
+          root.setAttribute('tabindex', '-1');
+        }
+        root.focus();
+      } else {
+        if (!this.props.allowFocusRoot) {
+          if (this._parkedTabIndex !== null) {
+            root.setAttribute('tabindex', this._parkedTabIndex);
+          } else {
+            root.removeAttribute('tabindex');
+          }
+        }
+      }
+    }
+  }
+
+  private _getElementByIndexPath(path: number[]): HTMLElement | undefined {
```

Comment:
I think just a brief comment here for the purpose of this function would help too (or a couple of statements within the function explaining what purpose this function serves.)

Reply:
Definitely!